unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BuildModeHUD/Scripts/BuilderInWorldLoadingController.cs

```diff
@@ -0,0 +1,35 @@
+public interface IBuilderInWorldLoadingController
+{
+    event System.Action OnCancelLoading;
+
+    void Initialize(IBuilderInWorldLoadingView initialLoadingView);
+    void Dispose();
+    void Show(bool showTips = true);
+    void Hide(bool forzeHidding = false);
+    void CancelLoading();
+}
+
+public class BuilderInWorldLoadingController : IBuilderInWorldLoadingController
```

Comment:
What do you think about making it inherit from `BIWController` like other controllers, it will create an monobehaviour but there is some functionality in the BuilderinworldController that can be implemented here with the references.

This way BuilderInWorldController doesn't touch anything from the loading, and this class control everything. Just to have the same structure for the BuilderInWrold

Reply:
I see this loading more as a HUD that have a view that is already a monobehaviour (`BuilderInWorldLoadingView`) and a controller that should only give orders to that view, and this controller is called by any consumer that needs it (in this case the `BuilderInWorldController` itself).

If we would want to extract logic from the `BuilderInWorldController` to the loading controller, for me the correct way would be to inject the dependency we need and then handle it as we want. But anyway, in terms of scalability, I still see that it makes more sense to treat the loading with the current philosophy of "it's just a HUD that is showed/hidden by any consumer that needs it, rather than putting specific BiW core logic in its controller.